cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -309,6 +309,14 @@ def get_ceph_external_mon_data(self):
         if len(q_leader_matching_list) == 0:
             raise ExecutionFailureException("No matching 'mon' details found")
         q_leader_details = q_leader_matching_list[0]
+        # get the address vector of the quorum-leader
+        q_leader_addrvec = q_leader_details.get('public_addrs', {}).get('addrvec', [])
+        # if the quorum-leader has only one address in the address-vector
+        # and it is of type 'v2' (ie; with <IP>:3300),
+        # raise an exception to make user aware that
+        # they have to enable 'v1' (ie; with <IP>:6789) type as well
+        if len(q_leader_addrvec) == 1 and q_leader_addrvec[0]['type'] == 'v2':
+            raise ExecutionFailureException("Only 'v2' address type is enabled, user should also enable 'v1' type as well")
```

Comment:
I am not sure whether this issue is related to below :

Rook assumes mon v2 always listen to `3300` and port parsed here to be `v1`. The ceph-mon can use different ports in `v1` and `v2`. And scripts always fetch the first port from the `ceph-mon dump -f` command it can be either v2 or v1.
Please refer [PopulateMonHostMembers()](https://github.com/rook/rook/blob/master/pkg/daemon/ceph/client/config.go#L251) for details.
Sorry I didn't file any Github issue till now  for the above issue
IMO script need to fetch both ports and Rook need to set them properly if both are passed to it.

Reply:
It's a different issue @thotz but we could certainly fix it in another PR I suppose by reading the port. But there is work to do in `PopulateMonHostMembers()` for sure.